File: M3Tools/Types/Events/CustomerEvents.vb
Namespace Events.Customers
	Public Delegate Sub CustomerEventHandler(sender As Object, e As CustomerEventArgs)
	'Public Delegate Sub EditCustomerEventHandler(sender As Object, e As CustomerEventArgs)

	Public Class CustomerEventArgs
		Inherits BaseArgs

		Property Customer As Types.Customer

		Sub New(customer As Types.Customer, Optional eventType As EventType = Nothing)
			Me.Customer = customer
			Me.EventType = eventType
		End Sub
	End Class
End Namespace

File: M3Tools/Types/Events/CustomerEvents.cs
namespace SPPBC.M3Tools.Events.Customers
{
	/// <summary>
	/// Event handler for when a customer based data event occurs
	/// </summary>
	/// <param name="sender"></param>
	/// <param name="e"></param>
	public delegate void CustomerEventHandler(object sender, CustomerEventArgs e);

	/// <summary>
	/// 
	/// </summary>
	sealed public class CustomerEventArgs : DataEventArgs<Types.Customer>
    {
		/// <summary>
		/// 
		/// </summary>
		public override Types.Customer Value { get; protected set; }

		/// <summary>
		/// 
		/// </summary>
		/// <param name="value"></param>
		/// <param name="type"></param>
		public CustomerEventArgs(Types.Customer value, EventType type = EventType.None) : base(value, type)
		{
		}
    }
}
